controllers/windowsmachine_controller.go

```diff
@@ -279,12 +281,6 @@ func (r *WindowsMachineReconciler) Reconcile(ctx context.Context, request ctrl.R
 			return ctrl.Result{}, nil
 		}
 	} else if *machine.Status.Phase != provisionedPhase {
-		log.V(1).Info("machine not provisioned", "phase", *machine.Status.Phase)
-		// configure Prometheus when a machine is not in `Running` or `Provisioned` phase. This configuration is
-		// required to update Endpoints object when Windows machines are being deleted.
-		if err := r.prometheusNodeConfig.Configure(); err != nil {
-			return ctrl.Result{}, errors.Wrap(err, "unable to configure Prometheus")
-		}
```

Comment:
It is unclear why you are removing this. Please explain.

Reply:
At this point,  triggering the Prometheus configuration when the machine is not in `Running` or `Provisioned` phases (i.e. Failed, Provisioning and Deleting phases) is not quite accurate, since the `Deleting` phase could be missed leaving the `windows-exporter`  endpoint with invalid information about the already deleted machine. The latter, was detected as a flake in the e2e test for vSphere.

In order to correct this issue, the Prometheus configuration is triggered when [the machine is not found](https://github.com/jrvaldes/windows-machine-config-operator/blob/b81c2f98c3c27dc239daabbbec9015e3bf6f6e4b/controllers/windowsmachine_controller.go#L228)